src/inmanta/deploy/scheduler.py

```diff
@@ -914,9 +921,7 @@ async def _new_version(
             self._timer_manager.remove_timers(deleted)
             # Install timers for initial up-to-date resources. They are up-to-date now,
             # but we want to make sure we periodically repair them.
-            self._timer_manager.update_timers(
-                up_to_date_resources | (transitive_unblocked - self._state.dirty), are_compliant=True
-            )
+            self._timer_manager.update_timers(up_to_date_resources | (transitive_unblocked - self._state.dirty))
```

Comment:
This suffices under the assumption that timers are already running for all other managed resources (not in the dirty set). This property is expected to hold during normal operation, but it might not when we restore state from a previous scheduler. i.e. we restore state, but we don't currently restore the timers. By restoring state, we cause some resources to be considered unchanged, i.e. we don't update their timer, hence we don't start it.

Concretely, I think we need to start them in the `_initialize` method. After calling `read_version()`, we could simply acquire the scheduler lock again, and call `update_timers(self._state.resources.keys() - self._state.dirty)`. I think that should suffice, but I'd appreciate if you could check the reasoning.

Reply:
I think you are right. 

On the other initialization path, I added the code to read_version